api_app/connectors_manager/serializers.py

```diff
@@ -13,6 +13,49 @@
 }
 
 
+class BaseField(serializers.Field):
+    def to_representation(self, value):
+        return value
+
+    def to_internal_value(self, data):
+        return data
+
+
+class SecretSerializer(serializers.Serializer):
+    """
+    validation serializer for `secrets` of ConnectorConfigSerializer
+    """
+
+    TYPE_CHOICES = (
+        ("number", "number"),
+        ("string", "string"),
+        ("bool", "bool"),
+    )
+
+    key_name = None
+
+    secret_name = serializers.CharField(max_length=128)
+    type = serializers.ChoiceField(choices=TYPE_CHOICES)
+    required = serializers.BooleanField()
+    default = BaseField(allow_null=True, required=True)
+    description = serializers.CharField(max_length=512)
+
+    def validate(self, data):
+        default, secret_type = data["default"], data["type"]
+        if default is not None and type(default) is not type(secret_type):
+            validation_error = {
+                self.key_name: {
+                    "default": f"should be of type {secret_type}, got {type(default)}"
+                }
+            }
+            raise serializers.ValidationError(validation_error)
+        return data
+
+    def to_internal_value(self, data):
```

Comment:
if done this way: https://github.com/intelowlproject/IntelOwl/pull/499/commits/d9b1872e344d58f76c454106af7fbffe1498d2dc#r656200927

then no need for handling tuple.